exec/java-exec/src/main/java/org/apache/drill/exec/physical/config/UnorderedMuxExchange.java

```diff
@@ -37,14 +37,8 @@ public UnorderedMuxExchange(@JsonProperty("child") PhysicalOperator child) {
   }
 
   @Override
-  public Receiver getReceiver(int minorFragmentId) {
-    createSenderReceiverMapping();
-
-    List<MinorFragmentEndpoint> senders = receiverToSenderMapping.get(minorFragmentId);
-    if (senders == null || senders.size() <= 0) {
-      throw new IllegalStateException(String.format("Failed to find senders for receiver [%d]", minorFragmentId));
-    }
-
+  protected Receiver getReceiverInternal(int oppositeMajorFragmentId,
+                                         List<MinorFragmentEndpoint> senders, boolean spooling) {
     return new UnorderedReceiver(senderMajorFragmentId, senders, false);
```

Comment:
`spooling` parameter is ignored, is this expected?

Reply:
Consider creating helper method like `getSenders()` that will return ` List<MinorFragmentEndpoint>` instead of `getReceiverInternal()`.

Reply:
Done.